apps/rush-lib/src/logic/taskRunner/ProjectBuilder.ts

```diff
@@ -227,15 +227,17 @@ export class ProjectBuilder extends BaseBuilder {
             files,
             arguments: this._commandToRun
           };
-        } else {
+        } else if (this.isIncrementalBuildAllowed) {
           terminal.writeLine(
             'Unable to calculate incremental build state. Instead running full rebuild. Ensure Git is present.'
           );
         }
       } catch (error) {
-        terminal.writeLine(
-          'Error calculating incremental build state. Instead running full rebuild. ' + error.toString()
-        );
+        if (this.isIncrementalBuildAllowed) {
+          terminal.writeLine(
+            'Error calculating incremental build state. Instead running full rebuild. ' + error.toString()
+          );
+        }
       }
 
       const isPackageUnchanged: boolean = !!(
```

Comment:
This variable is only used inside the `if (this.isIncrementalBuildAllowed) { ... }` block. Can this get moved into that block? Maybe even only after we try to restore from cache?